sw/airborne/modules/rot_wing_drone/rot_wing_automation.c

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2023 Dennis van Wijngaarden <D.C.vanWijngaarden@tudelft.nl>
+ *
+ * This file is part of paparazzi
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+/** @file "modules/rot_wing_drone/rot_wing_automation.c"
+ * @author Dennis van Wijngaarden <D.C.vanWijngaarden@tudelft.nl>
+ * Fucntions to automate the navigation and guidance of the rotating wing drone
+ */
+
+#include "modules/rot_wing_drone/rot_wing_automation.h"
+#include "state.h"
+#include "modules/datalink/telemetry.h"
+
+#ifndef ROT_WING_AUTOMATION_TRANS_ACCEL
+#define ROT_WING_AUTOMATION_TRANS_ACCEL 1.0
+#endif
+
+#ifndef ROT_WING_AUTOMATION_TRANS_DECEL
+#define ROT_WING_AUTOMATION_TRANS_DECEL 0.5
+#endif
+
+#ifndef ROT_WING_AUTOMATION_TRANS_LENGTH
+#define ROT_WING_AUTOMATION_TRANS_LENGTH 200.0
+#endif
+
+#ifndef ROT_WING_AUTOMATION_TRANS_AIRSPEED
+#define ROT_WING_AUTOMATION_TRANS_AIRSPEED 15.0
+#endif
+
+struct rot_wing_automation rot_wing_a;
+
+// declare function
+inline void update_waypoint_rot_wing_automation(uint8_t wp_id, struct FloatVect3 * target_ned);
+
+void init_rot_wing_automation(void)
+{
+  rot_wing_a.trans_accel = ROT_WING_AUTOMATION_TRANS_ACCEL;
+  rot_wing_a.trans_decel = ROT_WING_AUTOMATION_TRANS_DECEL;
+  rot_wing_a.trans_length = ROT_WING_AUTOMATION_TRANS_LENGTH;
+  rot_wing_a.trans_airspeed = ROT_WING_AUTOMATION_TRANS_AIRSPEED;
+  rot_wing_a.transitioned = false;
+}
+
+// periodic function
+void periodic_rot_wing_automation(void)
+{
+  float airspeed = stateGetAirspeed_f();
+  if (airspeed > rot_wing_a.trans_airspeed)
+  {
+    rot_wing_a.transitioned = true;
+  } else {
+    rot_wing_a.transitioned = false;
+  }
+}
+
+// Update a waypoint such that you can see on the GCS where the drone wants to go
+void update_waypoint_rot_wing_automation(uint8_t wp_id, struct FloatVect3 * target_ned) {
+
+  // Update the waypoint
+  struct EnuCoor_f target_enu;
+  ENU_OF_TO_NED(target_enu, *target_ned);
+  waypoint_set_enu(wp_id, &target_enu);
```

Comment:
It is required to add `#include "modules/nav/waypoints.h"` to pass the compilation tests.